### DISCUSSION TITLE
Combat system, targeting, etc.

### DISCUSSION THREAD

**Discussion by DiedByDisgust** [rogerdv/protoa#9] [Ideas]
Weapon properties proposal:
- weapon max range: where (<1m) is melee, and projectiles decay at max range

- melee weapon wide attack: where short, one handed, unarmed attacks targets only a single enemy, but longswords, greathammers, and other two handed large weapons can damage all enemies on front

### On (auto)attack order
If melee:
- take navpath to target 
- when gets at melee distance perform attack
- make damage to target( or targets(multiple) if wide melee weapon is used)

If range:
- take navpath to a point where there is no obstacles between player and target
- emit a projectile
- when projectile hits an enemy(whether intended target or not) makes damage
- projectile disappear on hit

### Target management
Single melee
if not at melee distance
![image](https://user-images.githubusercontent.com/25803026/203181783-3e8cd7e8-c466-4e99-a0fc-8529841e3ea7.png)

navpath until melee distance.
![image](https://user-images.githubusercontent.com/25803026/203182389-0e114ed6-0e3e-4956-8ac5-0f438492786a.png)

if wide melee
![image](https://user-images.githubusercontent.com/25803026/203183734-006f22a9-2561-4ea9-b7b2-ffecd1193239.png)

with range weapon aiming
![image](https://user-images.githubusercontent.com/25803026/203219184-97ad486c-ce98-466a-9aac-f9e4a6238ed4.png)

### Damage management
**melee**
basically faked. dividing the process into a visual component and a back end for the interaction. to avoid collision conflicts during the melee simlations. 
- Collision only for animation purposes like bleeding and others
- Real damage managed through back end.
**ranged**
rely completely on collision of the projectile. no faking.
